WBooks/Screens/BookDetails/BookDetailsViewController.swift

```diff
@@ -0,0 +1,126 @@
+    public override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        fatalError("init(nibName:bundle:) has not been implemented")
+    }
+    
+    // MARK: - BookDetailsViewController
+
+    init(viewModel: BookDetailsViewModel) {
+        _bookDetailsViewModel = viewModel
+        super.init(nibName: .none, bundle: .none)
+    }
+    
+    override public func loadView() {
+        view = _view
+        _view.bookTitle.text = _bookDetailsViewModel.bookModel.title
+        _view.bookGen.text = _bookDetailsViewModel.bookModel.genre
+        _view.bookYear.text = _bookDetailsViewModel.bookModel.year
+        _view.bookAuthor.text = _bookDetailsViewModel.bookModel.author
+        _view.bookStatus.text = _bookDetailsViewModel.bookModel.status
+        _view.updateStyles(newStatus: _bookDetailsViewModel.bookModel.status)
+        if _bookDetailsViewModel.bookModel.image != nil {
+            _view.bookImage.loadUrl(from: _bookDetailsViewModel.bookModel.image ?? "")
+        } else {
+            _view.bookImage.image = UIImage.defaultBook
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        _view.updateCommentsLoading(isLoading: true)
+        _bookDetailsViewModel.getBookComments(onSuccess: getCommentsSuccess, onError: getCommentsError)
+        view.layer.backgroundColor = UIColor.creamBlue?.cgColor
+        setTitle(headerTitle: "BOOK_DETAILS_HEADER_TITLE".localized())
+        setNavigationBackButton()
+        _view.rentButton.addTarget(self, action: #selector(rentButtonTapped), for: .touchUpInside)
+    }
+    
+    func setupTableView() {
+        _view.commentsTable.delegate = self
+        _view.commentsTable.dataSource = self
+        _view.commentsTable.register(cell: CommentCell.self)
+        _view.commentsTable.separatorStyle = .none
+    }
+    
+    @objc private func rentButtonTapped() {
+        if _bookDetailsViewModel.bookModel.status == "Unavailable" {
+            let alert = UIAlertController().createErrorAlert(message: "RENT_UNAVAILABLE".localized())
+            self.present(alert, animated: true)
+        } else {
+            _view.updateRentLoading(isLoading: true)
+            _bookDetailsViewModel.rentBook(onSuccess: rentBookSuccess, onError: rentBookError)
+        }
+    }
+}
+
+// MARK: - Extension for request callbacks
+extension BookDetailsViewController {
+    func rentBookSuccess() {
+        _view.bookStatus.text = "Unavailable"
+        _view.updateStyles(newStatus: "Unavailable")
+        _view.updateRentLoading(isLoading: false)
+    }
+    
+    func rentBookError() {
+        let alert = UIAlertController().createErrorAlert(message: "RENT_ERROR".localized())
+        self.present(alert, animated: true)
+        _view.updateRentLoading(isLoading: false)
+    }
+    
+    func getCommentsSuccess() {
+        _view.updateCommentsLoading(isLoading: false)
+        if _bookDetailsViewModel.numberOfComments == 0 {
```

Comment:
you could have a property in viewModel that validates this, using .isEmpty (not .count)